react/features/facial-recognition/actions.js

```diff
@@ -171,73 +208,30 @@ export function startFacialRecognition() {
  * @returns {void}
  */
 export function stopFacialRecognition() {
-    return function(dispatch: Function, getState: Function) {
-        const state = getState();
-        const { recognitionActive } = state['features/facial-recognition'];
-
-        if (!recognitionActive) {
-            imageCapture = null;
-
-            return;
-        }
-        imageCapture = null;
-        worker.postMessage({
-            type: CLEAR_TIMEOUT
-        });
-
+    return function(dispatch: Function) {
         if (lastFacialExpression && lastFacialExpressionTimestamp) {
             dispatch(
-                        addFacialExpression(
-                            lastFacialExpression,
-                            duplicateConsecutiveExpressions + 1,
-                            lastFacialExpressionTimestamp
-                        )
+                addFacialExpression(
+                    lastFacialExpression,
+                    duplicateConsecutiveExpressions + 1,
+                    lastFacialExpressionTimestamp
+                )
             );
         }
+
+        clearInterval(webhookSendInterval);
```

Comment:
I think we should use a single interval and perform multiple detections in a single step.

Reply:
not sure what you mean, one interval is used for sending frames to worker, the other one is for sending collected expressions on a webhook

Reply:
Sorry, I realize I left the comment in the wrong place. I see you've made all landmark detections work on a single timer. Is that something desirable long term? Do we need to detect emotions with the same frequency as centering?

Reply:
ideally we would do both detections for a given frame, so we can either use the min or the max interval from the two. Currently the minimum is set.

Reply:
Maybe we can revisit in the future, if any of the landmarks poses a problem. WDYT @gabiborlea ?